src/foundry/foundry.js/clientDocuments/scene.d.ts

```diff
@@ -8,9 +18,450 @@ declare global {
+      options: DocumentModificationContext,
+      userId: string
+    ): void;
+    _onUpdateEmbeddedDocuments(
+      embeddedName: string,
+      documents: InstanceType<ConfiguredDocumentClass<typeof foundry.documents.BaseWall>>[], // TODO Replace with WallDocument
+      result: DeepPartial<PropertiesToSource<WallDataProperties>>[],
+      options: DocumentModificationContext,
+      userId: string
+    ): void;
+
+    /** @override */
+    _preDeleteEmbeddedDocuments(
+      embeddedName: string,
+      result: string[],
+      options: DocumentModificationContext,
+      userId: string
+    ): void;
+
+    /** @override */
+    _onDeleteEmbeddedDocuments(
+      embeddedName: string,
+      documents: InstanceType<ConfiguredDocumentClass<typeof DrawingDocument>>[],
+      result: string[],
+      options: DocumentModificationContext,
+      userId: string
+    ): void;
+    _onDeleteEmbeddedDocuments(
+      embeddedName: string,
+      documents: InstanceType<ConfiguredDocumentClass<typeof TokenDocument>>[],
+      result: string[],
+      options: DocumentModificationContext,
+      userId: string
+    ): void;
+    _onDeleteEmbeddedDocuments(
+      embeddedName: string,
+      documents: InstanceType<ConfiguredDocumentClass<typeof AmbientLightDocument>>[],
+      result: string[],
+      options: DocumentModificationContext,
+      userId: string
+    ): void;
+    _onDeleteEmbeddedDocuments(
+      embeddedName: string,
+      documents: InstanceType<ConfiguredDocumentClass<typeof foundry.documents.BaseNote>>[], // TODO Replace with NoteDocument
+      result: string[],
+      options: DocumentModificationContext,
+      userId: string
+    ): void;
+    _onDeleteEmbeddedDocuments(
+      embeddedName: string,
+      documents: InstanceType<ConfiguredDocumentClass<typeof foundry.documents.BaseAmbientSound>>[], // TODO Replace with AmbientSoundDocument
+      result: string[],
+      options: DocumentModificationContext,
+      userId: string
+    ): void;
+    _onDeleteEmbeddedDocuments(
+      embeddedName: string,
+      documents: InstanceType<ConfiguredDocumentClass<typeof MeasuredTemplateDocument>>[],
+      result: string[],
+      options: DocumentModificationContext,
+      userId: string
+    ): void;
+    _onDeleteEmbeddedDocuments(
+      embeddedName: string,
+      documents: InstanceType<ConfiguredDocumentClass<typeof foundry.documents.BaseTile>>[], // TODO Replace with TileDocument
+      result: string[],
+      options: DocumentModificationContext,
+      userId: string
+    ): void;
+    _onDeleteEmbeddedDocuments(
+      embeddedName: string,
+      documents: InstanceType<ConfiguredDocumentClass<typeof foundry.documents.BaseWall>>[], // TODO Replace with WallDocument
+      result: string[],
+      options: DocumentModificationContext,
+      userId: string
+    ): void;
+
+    // TODO: Improve when `pack` is typed (see Document.toCompendium) 9716
```

Comment:
What exactly should be improved here? I think it's not actually worth doing so we should probably just get rid of the TODO here and in `ClientDocumentMixin`.